scripts/zones/Arrapago_Reef/Zone.lua

```diff
@@ -34,6 +35,18 @@ end
 zoneObject.afterZoneIn = function(player)
     player:entityVisualPacket('1pb1')
     player:entityVisualPacket('2pb1')
+    -- Add Lamia No.19 death listener
+    player:addListener('DEATH', 'LAMIA_NO_19_PLAYERDEATH', function(playerArg)
+        local ID = zones[xi.zone.ARRAPAGO_REEF]
+        local lamiaMobId = ID.mob.LAMIA_NO19
+        local x = playerArg:getXPos()
+        local y = playerArg:getYPos()
+        local z = playerArg:getZPos()
+        local mob = SpawnMob(lamiaMobId)
+        if mob then
+            mob:setPos(x, y, z)
+        end
```

Comment:
Recommend to also move the zone ID here to the top of the file to keep it styled like other files. It would go above TZone line. Can take a look at other Zone files to see typically.

We can also shrink this section a bit and set the mob's spawn position before its spawned:
```suggestion
        local lamiaMob = GetMobByID(ID.mob.LAMIA_NO19)
        if lamiaMob then
            lamiaMob:setSpawn(playerArg:getXPos(), playerArg:getYPos(), playerArg:getZPos())
            lamiaMob:spawn()
        end
```